From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Tue, 13 May 2025 15:53:41 +0000
Subject: [PATCH] Search engine data patches to always include Mullvad Leta and
 DuckDuckGo

These changes allow inclusion of Mullvad Leta search engine to offered
list of search engines on all regions, along with the current default,
DuckDuckGo.
---
 ...l_settings.json-search-engines-secti.patch |  140 ++
 ...llvad-Leta-to-list-of-search-engines.patch |   29 +
 ...ent-maximum-number-of-search-engines.patch |   34 +
 ...ment-version-for-search-engines-data.patch |   34 +
 ...nd-Mullvad-Leta-on-all-regional-list.patch | 1353 +++++++++++++++++
 5 files changed, 1590 insertions(+)
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Reformat-regional_settings.json-search-engines-secti.patch
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Add-Mullvad-Leta-to-list-of-search-engines.patch
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-maximum-number-of-search-engines.patch
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Increment-version-for-search-engines-data.patch
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0005-Add-DuckDuckGo-and-Mullvad-Leta-on-all-regional-list.patch

diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Reformat-regional_settings.json-search-engines-secti.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Reformat-regional_settings.json-search-engines-secti.patch
new file mode 100644
index 0000000000000..ee609d40b7ad4
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Reformat-regional_settings.json-search-engines-secti.patch
@@ -0,0 +1,140 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Mon, 29 Sep 2025 11:48:26 +0000
+Subject: [PATCH] Reformat regional_settings.json search engines section
+
+Commands used for generating the format changes:
+
+```bash
+    sed -ri "s:\[\":\[\n        \":g" definitions/regional_settings.json
+
+    sed -ri "s:\"\]:\"\n      \]:g" definitions/regional_settings.json
+
+    sed -ri "s:\, \":\,\n        \":g" definitions/regional_settings.json
+
+```bash
+---
+ definitions/regional_settings.json | 62 ++++++++++++++++++++++++++----
+ 1 file changed, 54 insertions(+), 8 deletions(-)
+
+diff --git a/definitions/regional_settings.json b/definitions/regional_settings.json
+index 9bd4f39..7bdc719 100644
+--- a/definitions/regional_settings.json
++++ b/definitions/regional_settings.json
+@@ -194,7 +194,13 @@
+     },
+     "CN": {
+       // China
+-      "search_engines": ["&baidu", "&bing", "&sogou", "&so_360", "&google"]
++      "search_engines": [
++        "&baidu",
++        "&bing",
++        "&sogou",
++        "&so_360",
++        "&google"
++      ]
+     },
+     "CO": {
+       // Colombia
+@@ -405,7 +411,13 @@
+     },
+     "HK": {
+       // Hong Kong
+-      "search_engines": ["&google", "&bing", "&yahoo_hk", "&baidu", "&so_360"]
++      "search_engines": [
++        "&google",
++        "&bing",
++        "&yahoo_hk",
++        "&baidu",
++        "&so_360"
++      ]
+     },
+     "HN": {
+       // Honduras
+@@ -574,7 +586,13 @@
+     },
+     "KR": {
+       // South Korea
+-      "search_engines": ["&google", "&naver", "&bing", "&daum", "&coccoc"]
++      "search_engines": [
++        "&google",
++        "&naver",
++        "&bing",
++        "&daum",
++        "&coccoc"
++      ]
+     },
+     "KW": {
+       // Kuwait
+@@ -588,7 +606,13 @@
+     },
+     "KZ": {
+       // Kazakhstan
+-      "search_engines": ["&google", "&yandex_kz", "&bing", "&mail_ru", "&yahoo"]
++      "search_engines": [
++        "&google",
++        "&yandex_kz",
++        "&bing",
++        "&mail_ru",
++        "&yahoo"
++      ]
+     },
+     "LB": {
+       // Lebanon
+@@ -1061,7 +1085,13 @@
+     },
+     "TW": {
+       // Taiwan
+-      "search_engines": ["&google", "&yahoo_tw", "&bing", "&coccoc", "&baidu"]
++      "search_engines": [
++        "&google",
++        "&yahoo_tw",
++        "&bing",
++        "&coccoc",
++        "&baidu"
++      ]
+     },
+     "TZ": {
+       // Tanzania
+@@ -1115,7 +1145,13 @@
+     },
+     "VN": {
+       // Vietnam
+-      "search_engines": ["&google", "&coccoc", "&yahoo", "&bing", "&duckduckgo"]
++      "search_engines": [
++        "&google",
++        "&coccoc",
++        "&yahoo",
++        "&bing",
++        "&duckduckgo"
++      ]
+     },
+     "YE": {
+       // Yemen
+@@ -1139,14 +1175,24 @@
+     },
+     "ZW": {
+       // Zimbabwe
+-      "search_engines": ["&google", "&bing", "&yahoo", "&duckduckgo", "&sogou"]
++      "search_engines": [
++        "&google",
++        "&bing",
++        "&yahoo",
++        "&duckduckgo",
++        "&sogou"
++      ]
+     },
+     // Fallback / default country. Occupies a user-assignable alpha-2 code,
+     // representing unknown or invalid territory.
+     // See https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#ZZ
+     "ZZ": {
+       // Default
+-      "search_engines": ["&google", "&bing", "&yahoo"]
++      "search_engines": [
++        "&google",
++        "&bing",
++        "&yahoo"
++      ]
+     }
+   },
+   "aggregation": {
diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Add-Mullvad-Leta-to-list-of-search-engines.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Add-Mullvad-Leta-to-list-of-search-engines.patch
new file mode 100644
index 0000000000000..ae4ee675af301
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Add-Mullvad-Leta-to-list-of-search-engines.patch
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:41:50 +0000
+Subject: [PATCH] Add Mullvad Leta to list of search engines
+
+---
+ definitions/prepopulated_engines.json | 9 +++++++++
+ 1 file changed, 9 insertions(+)
+
+diff --git a/definitions/prepopulated_engines.json b/definitions/prepopulated_engines.json
+index 35ad4c0..c05f9ad 100644
+--- a/definitions/prepopulated_engines.json
++++ b/definitions/prepopulated_engines.json
+@@ -278,6 +278,15 @@
+       "id": 103
+     },
+ 
++    "mullvad_leta": {
++      "name": "Mullvad Leta",
++      "keyword": "leta.mullvad.net",
++      "favicon_url": "https://leta.mullvad.net/favicon.ico",
++      "search_url": "https://leta.mullvad.net/search?q={searchTerms}",
++      "type": "SEARCH_ENGINE_MULLVAD_LETA",
++      "id": 116
++    },
++
+     "naver": {
+       "name": "\ub124\uc774\ubc84",
+       "keyword": "naver.com",
diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-maximum-number-of-search-engines.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-maximum-number-of-search-engines.patch
new file mode 100644
index 0000000000000..3bdabfa66d19f
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-maximum-number-of-search-engines.patch
@@ -0,0 +1,34 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:41:06 +0000
+Subject: [PATCH] Increment maximum number of search engines
+
+Commands used for incrementing version:
+
+```bash
+    version_line=$(echo $(grep kCurrentDataVersion definitions/prepopulated_engines.json))
+
+    version=$(echo $version_line | sed -r "s/.*\"kMaxPrepopulatedEngineID\": //g")
+
+    incremented_version=$(($version + 1)); incremented_version_line=$(echo $version_line | sed -r "s/$version/$incremented_version/g")
+
+    sed -ri "s/$version_line/$incremented_version_line/g" definitions/prepopulated_engines.json
+
+```
+---
+ definitions/prepopulated_engines.json | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/definitions/prepopulated_engines.json b/definitions/prepopulated_engines.json
+index c05f9ad..c2a68e9 100644
+--- a/definitions/prepopulated_engines.json
++++ b/definitions/prepopulated_engines.json
+@@ -22,7 +22,7 @@
+     // use a lower number. This number is used by UMA stats and must be updated
+     // to include the full range or else not all engines will be counted in
+     // stats.
+-    "kMaxPrepopulatedEngineID": 115,
++    "kMaxPrepopulatedEngineID": 116,
+ 
+     // Increment this if you change the data in ways that mean users with
+     // existing data should get a new version. Otherwise, existing data may
diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Increment-version-for-search-engines-data.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Increment-version-for-search-engines-data.patch
new file mode 100644
index 0000000000000..a0c3470a59f6e
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Increment-version-for-search-engines-data.patch
@@ -0,0 +1,34 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:41:30 +0000
+Subject: [PATCH] Increment version for search engines data
+
+Commands used for incrementing version:
+
+```bash
+    version_line=$(echo $(grep kCurrentDataVersion definitions/prepopulated_engines.json))
+
+    version=$(echo $version_line | sed -r "s/.*\"kCurrentDataVersion\": //g")
+
+    incremented_version=$(($version + 1)); incremented_version_line=$(echo $version_line | sed -r "s/$version/$incremented_version/g")
+
+    sed -ri "s/$version_line/$incremented_version_line/g" definitions/prepopulated_engines.json
+
+```
+---
+ definitions/prepopulated_engines.json | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/definitions/prepopulated_engines.json b/definitions/prepopulated_engines.json
+index c2a68e9..9a97180 100644
+--- a/definitions/prepopulated_engines.json
++++ b/definitions/prepopulated_engines.json
+@@ -28,7 +28,7 @@
+     // existing data should get a new version. Otherwise, existing data may
+     // continue to be used and updates made here will not always appear.
+     // Also then run tools/search_engine_choice/generate_search_engine_icons.py.
+-    "kCurrentDataVersion": 191
++    "kCurrentDataVersion": 192
+   },
+ 
+   // The following engines are included in country lists and are added to the
diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0005-Add-DuckDuckGo-and-Mullvad-Leta-on-all-regional-list.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0005-Add-DuckDuckGo-and-Mullvad-Leta-on-all-regional-list.patch
new file mode 100644
index 0000000000000..d4ffd6df032df
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0005-Add-DuckDuckGo-and-Mullvad-Leta-on-all-regional-list.patch
@@ -0,0 +1,1353 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Mon, 29 Sep 2025 12:15:48 +0000
+Subject: [PATCH] Add DuckDuckGo and Mullvad Leta on all regional list of
+ search engines
+
+Commands used for appending and replacing regional list of search
+engines json, in order:
+
+```bash
+    sed -ri "N;N;s/,(|\n)?\n        \"\&duckduckgo\"//g;P;D" definitions/regional_settings.json
+
+    sed -ri "s/\"search_engines\": \[/&\n        \"\&duckduckgo\",\n        \"\&mullvad_leta\",/g" definitions/regional_settings.json
+
+```
+---
+ definitions/regional_settings.json | 378 +++++++++++++++++++----------
+ 1 file changed, 247 insertions(+), 131 deletions(-)
+
+diff --git a/definitions/regional_settings.json b/definitions/regional_settings.json
+index 7bdc719..0bac9c4 100644
+--- a/definitions/regional_settings.json
++++ b/definitions/regional_settings.json
+@@ -6,39 +6,43 @@
+     "AE": {
+       // United Arab Emirates
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_ru",
+-        "&duckduckgo"
++        "&yandex_ru"
+       ]
+     },
+     "AL": {
+       // Albania
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_tr",
+-        "&duckduckgo"
++        "&yandex_tr"
+       ]
+     },
+     "AR": {
+       // Armenia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_ar",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "AT": {
+       // Austria
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&ecosia",
+         "&bing",
+         "&yahoo_at",
+@@ -49,29 +53,32 @@
+     "AU": {
+       // Australia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_au",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "BA": {
+       // Bosnia and Herzegovina
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "BE": {
+       // Belgium
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&ecosia",
+         "&bing",
+         "&qwant",
+@@ -82,9 +89,10 @@
+     "BG": {
+       // Bulgaria
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&yep",
+         "&ecosia",
+@@ -95,69 +103,76 @@
+     "BH": {
+       // Bahrain
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "BI": {
+       // Burundi
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_ru",
+-        "&duckduckgo"
++        "&yandex_ru"
+       ]
+     },
+     "BN": {
+       // Brunei
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "BO": {
+       // Bolivia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "BR": {
+       // Brazil
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_br",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "BY": {
+       // Belarus
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yandex_by",
+         "&bing",
+-        "&duckduckgo",
+         "&mail_ru"
+       ]
+     },
+     "BZ": {
+       // Belize
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+-        "&duckduckgo",
+         "&yahoo",
+         "&yandex_com"
+       ]
+@@ -165,19 +180,21 @@
+     "CA": {
+       // Canada
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_ca",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "CH": {
+       // Switzerland
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+-        "&duckduckgo",
+         "&yahoo_ch",
+         "&ecosia"
+       ]
+@@ -185,16 +202,19 @@
+     "CL": {
+       // Chile
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_cl",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "CN": {
+       // China
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&baidu",
+         "&bing",
+         "&sogou",
+@@ -205,29 +225,32 @@
+     "CO": {
+       // Colombia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_co",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "CR": {
+       // Costa Rica
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "CY": {
+       // Republic of Cyprus
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&yahoo_emea",
+         "&ecosia",
+@@ -238,10 +261,11 @@
+     "CZ": {
+       // Czechia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&seznam",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&privacywall",
+@@ -251,8 +275,9 @@
+     "DE": {
+       // Germany
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&brave",
+         "&ecosia",
+         "&bing",
+@@ -264,9 +289,10 @@
+     "DK": {
+       // Denmark
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&qwant",
+@@ -277,39 +303,43 @@
+     "DO": {
+       // Dominican Republic
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "DZ": {
+       // Algeria
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_fr",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "EC": {
+       // Ecuador
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_es",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "EE": {
+       // Estonia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&qwant",
+@@ -320,20 +350,22 @@
+     "EG": {
+       // Egypt
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "ES": {
+       // Spain
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+         "&bing",
+-        "&duckduckgo",
+         "&ecosia",
+         "&yahoo_es",
+         "&privacywall",
+@@ -343,9 +375,10 @@
+     "FI": {
+       // Finland
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&qwant",
+@@ -356,21 +389,23 @@
+     "FO": {
+       // Faroe Islands
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_uk",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "FR": {
+       // France
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+         "&ecosia",
+         "&bing",
+-        "&duckduckgo",
+         "&qwant",
+         "&yahoo_fr",
+         "&lilo"
+@@ -379,19 +414,21 @@
+     "GB": {
+       // United Kingdom
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_uk",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "GR": {
+       // Greece
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&yahoo_emea",
+         "&ecosia",
+@@ -402,16 +439,19 @@
+     "GT": {
+       // Guatemala
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "HK": {
+       // Hong Kong
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_hk",
+@@ -422,19 +462,21 @@
+     "HN": {
+       // Honduras
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "HR": {
+       // Croatia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&yep",
+         "&ecosia",
+@@ -445,10 +487,11 @@
+     "HU": {
+       // Hungary
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+         "&bing",
+-        "&duckduckgo",
+         "&ecosia",
+         "&yep",
+         "&yahoo_emea",
+@@ -458,18 +501,20 @@
+     "ID": {
+       // Indonesia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yandex_com",
+         "&bing",
+-        "&yahoo_id",
+-        "&duckduckgo"
++        "&yahoo_id"
+       ]
+     },
+     "IE": {
+       // Ireland
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&brave",
+         "&bing",
+         "&yahoo_uk",
+@@ -481,48 +526,53 @@
+     "IL": {
+       // Israel
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yandex_ru",
+-        "&yahoo",
+-        "&duckduckgo"
++        "&yahoo"
+       ]
+     },
+     "IN": {
+       // India
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_in",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "IQ": {
+       // Iraq
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yandex_tr",
+-        "&yahoo",
+-        "&duckduckgo"
++        "&yahoo"
+       ]
+     },
+     "IR": {
+       // Iran
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "IS": {
+       // Iceland
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&brave",
+         "&bing",
+         "&ecosia",
+@@ -534,10 +584,11 @@
+     "IT": {
+       // Italy
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+         "&bing",
+-        "&duckduckgo",
+         "&ecosia",
+         "&yahoo_it",
+         "&qwant",
+@@ -547,46 +598,52 @@
+     "JM": {
+       // Jamaica
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "JO": {
+       // Jordan
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "JP": {
+       // Japan
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yahoo_jp",
+         "&bing",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "KE": {
+       // Kenya
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "KR": {
+       // South Korea
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&naver",
+         "&bing",
+@@ -597,16 +654,19 @@
+     "KW": {
+       // Kuwait
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yandex_tr",
+-        "&yahoo",
+-        "&duckduckgo"
++        "&yahoo"
+       ]
+     },
+     "KZ": {
+       // Kazakhstan
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yandex_kz",
+         "&bing",
+@@ -617,18 +677,20 @@
+     "LB": {
+       // Lebanon
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "LI": {
+       // Liechtenstein
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&brave",
+         "&bing",
+         "&ecosia",
+@@ -640,9 +702,10 @@
+     "LT": {
+       // Lithuania
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&yahoo_emea",
+@@ -653,9 +716,10 @@
+     "LU": {
+       // Luxembourg
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&qwant",
+@@ -666,9 +730,10 @@
+     "LV": {
+       // Latvia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&privacywall",
+@@ -679,70 +744,77 @@
+     "LY": {
+       // Libya
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yandex_tr",
+-        "&yahoo",
+-        "&duckduckgo"
++        "&yahoo"
+       ]
+     },
+     "MA": {
+       // Morocco
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_fr",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "MC": {
+       // Monaco
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_fr",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "MD": {
+       // Moldova
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yandex_ru",
+         "&bing",
+-        "&duckduckgo",
+         "&mail_ru"
+       ]
+     },
+     "ME": {
+       // Montenegro
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yandex_ru",
+-        "&yahoo",
+-        "&duckduckgo"
++        "&yahoo"
+       ]
+     },
+     "MK": {
+       // Macedonia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "MT": {
+       // Malta
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+         "&bing",
+-        "&duckduckgo",
+         "&ecosia",
+         "&yahoo_emea",
+         "&privacywall",
+@@ -752,38 +824,42 @@
+     "MX": {
+       // Mexico
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_mx",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "MY": {
+       // Malaysia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_my",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "NI": {
+       // Nicaragua
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "NL": {
+       // Netherlands
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&brave",
+         "&bing",
+         "&ecosia",
+@@ -795,8 +871,9 @@
+     "NO": {
+       // Norway
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&brave",
+         "&bing",
+         "&ecosia",
+@@ -808,69 +885,76 @@
+     "NZ": {
+       // New Zealand
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_nz",
+-        "&duckduckgo",
+         "&ecosia"
+       ]
+     },
+     "OM": {
+       // Oman
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "PA": {
+       // Panama
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_es",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "PE": {
+       // Peru
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_pe",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "PH": {
+       // Philippines
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_ph",
+-        "&ecosia",
+-        "&duckduckgo"
++        "&ecosia"
+       ]
+     },
+     "PK": {
+       // Pakistan
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "PL": {
+       // Poland
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&yahoo_emea",
+         "&ecosia",
+@@ -881,20 +965,22 @@
+     "PR": {
+       // Puerto Rico
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "PT": {
+       // Portugal
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+         "&bing",
+-        "&duckduckgo",
+         "&ecosia",
+         "&yep",
+         "&qwant",
+@@ -904,29 +990,32 @@
+     "PY": {
+       // Paraguay
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_es",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "QA": {
+       // Quatar
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_tr"
+       ]
+     },
+     "RO": {
+       // Romania
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&yahoo_emea",
+         "&bing",
+         "&yep",
+@@ -937,48 +1026,53 @@
+     "RS": {
+       // Serbia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&yandex_ru",
+-        "&duckduckgo"
++        "&yandex_ru"
+       ]
+     },
+     "RU": {
+       // Russia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&yandex_ru",
+         "&google",
+         "&bing",
+-        "&duckduckgo",
+         "&mail_ru"
+       ]
+     },
+     "RW": {
+       // Rwanda
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "SA": {
+       // Saudi Arabia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yandex_com",
+-        "&yahoo",
+-        "&duckduckgo"
++        "&yahoo"
+       ]
+     },
+     "SE": {
+       // Sweden
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&brave",
+         "&bing",
+         "&ecosia",
+@@ -990,19 +1084,21 @@
+     "SG": {
+       // Singapore
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_sg",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "SI": {
+       // Slovenia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&ecosia",
+         "&yep",
+@@ -1013,9 +1109,10 @@
+     "SK": {
+       // Slovakia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&brave",
+-        "&duckduckgo",
+         "&bing",
+         "&seznam",
+         "&ecosia",
+@@ -1026,66 +1123,74 @@
+     "SV": {
+       // El Salvador
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_es",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "SY": {
+       // Syria
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yandex_ru",
+-        "&yahoo",
+-        "&duckduckgo"
++        "&yahoo"
+       ]
+     },
+     "TH": {
+       // Thailand
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_th",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "TN": {
+       // Tunisia
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_fr",
+-        "&yandex_com",
+-        "&duckduckgo"
++        "&yandex_com"
+       ]
+     },
+     "TR": {
+       // Turkey
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yandex_tr",
+         "&yahoo_tr",
+-        "&bing",
+-        "&duckduckgo"
++        "&bing"
+       ]
+     },
+     "TT": {
+       // Trinidad and Tobago
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "TW": {
+       // Taiwan
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yahoo_tw",
+         "&bing",
+@@ -1096,8 +1201,9 @@
+     "TZ": {
+       // Tanzania
+       "search_engines": [
+-        "&google",
+         "&duckduckgo",
++        "&mullvad_leta",
++        "&google",
+         "&bing",
+         "&yahoo",
+         "&yandex_com"
+@@ -1106,80 +1212,88 @@
+     "UA": {
+       // Ukraine
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&yandex_ru",
+         "&bing",
+-        "&duckduckgo",
+         "&yahoo"
+       ]
+     },
+     "US": {
+       // United States
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "UY": {
+       // Uruguay
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_es",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "VE": {
+       // Venezuela
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo_es",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "VN": {
+       // Vietnam
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&coccoc",
+         "&yahoo",
+-        "&bing",
+-        "&duckduckgo"
++        "&bing"
+       ]
+     },
+     "YE": {
+       // Yemen
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "ZA": {
+       // South Africa
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&yandex_com"
+       ]
+     },
+     "ZW": {
+       // Zimbabwe
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo",
+-        "&duckduckgo",
+         "&sogou"
+       ]
+     },
+@@ -1189,6 +1303,8 @@
+     "ZZ": {
+       // Default
+       "search_engines": [
++        "&duckduckgo",
++        "&mullvad_leta",
+         "&google",
+         "&bing",
+         "&yahoo"
